### PR TITLE
PUBLIC_LAW - Add mapping rules in ORM for STAFF users

### PR DESCRIPTION
**Pull request by Karthikarenjith**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-2463

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/validationrules/publiclaw/publiclaw-admin-mapping.drl [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      .actorId($cap.getId())
+      .roleCategory(RoleCategory.ADMIN)
+      .roleType(RoleType.ORGANISATION)
+      .roleName("specific-access-approver-admin")
+      .grantType(GrantType.STANDARD)
+      .classification(Classification.PUBLIC)
+      .readOnly(false)
+      .attributes(attribute)
+      .build());
+      logMsg("Rule : publiclaw_specific-access-approver-admin_org_role");
```
Comment (sankar-padakula) [inline src/main/resources/validationrules/publiclaw/publiclaw-admin-mapping.drl:102]: please follow same naming convention on rule and logmsg. replace hyphen with underscore

File: src/main/resources/validationrules/publiclaw/publiclaw-caseworker-mapping.drl [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      .actorId($cap.getId())
+      .roleCategory(RoleCategory.LEGAL_OPERATIONS)
+      .roleType(RoleType.ORGANISATION)
+      .roleName("specific-access-approver-legal-ops")
+      .grantType(GrantType.STANDARD)
+      .classification(Classification.PUBLIC)
+      .readOnly(false)
+      .attributes(attribute)
+      .build());
+      logMsg("Rule : publiclaw_specific-access-approver-legal-ops_org_role");
```
Comment (sankar-padakula) [inline src/main/resources/validationrules/publiclaw/publiclaw-caseworker-mapping.drl:71]: here also same follow same naming convention

File: src/main/resources/validationrules/core/hearing-role-ctsc-global.drl [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ * For on-boarding new service, we just need to include the corresponding service code in every
+ * rule as per the hearing role requirements captured on service configuration page.
+ */
+
+/*
+ * All services "hearing-manager" Org role mapping.
+ */
+ rule "global_hearing_manager_ctsc"
+ when
+   $f:  FeatureFlag(status && flagName == FeatureFlagEnum.SSCS_HEARING_1_0.getValue())
```
Comment (sankar-padakula) [inline src/main/resources/validationrules/core/hearing-role-ctsc-global.drl:27]: hearing flag SSCS_HEARING_1_0 is already live. So can it go with PUBLICLAW_WA_1_0 or without flag?
@nitinsahu1

File: src/main/resources/validationrules/publiclaw/publiclaw-admin-mapping.drl [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+rule "publiclaw_hearing_centre_team_leader_org_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.PUBLICLAW_WA_1_0.getValue())
+  $cap: CaseWorkerAccessProfile(roleId =="3", serviceCode == "ABA3", !suspended)
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+   attribute.put("workTypes", JacksonUtils.convertObjectIntoJsonNode("routine_work,access_requests,hearing_work,applications"));
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-admin-mapping.drl:29]: according to confluence these work types should be
routine_work access_requests

File: src/main/resources/validationrules/publiclaw/publiclaw-admin-mapping.drl [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+   attribute.put("region", JacksonUtils.convertObjectIntoJsonNode($cap.getRegionId()));
+
+  insert(
+      RoleAssignment.builder()
+      .actorIdType(ActorIdType.IDAM)
+      .actorId($cap.getId())
+      .roleCategory(RoleCategory.ADMIN)
+      .roleType(RoleType.ORGANISATION)
+      .roleName("task-supervisor")
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-admin-mapping.drl:68]: this role requires a worktype of 
routine_work
 

<br class="Apple-interchange-newline">

File: src/main/resources/validationrules/publiclaw/publiclaw-admin-mapping.drl [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+   attribute.put("region", JacksonUtils.convertObjectIntoJsonNode($cap.getRegionId()));
+
+  insert(
+      RoleAssignment.builder()
+      .actorIdType(ActorIdType.IDAM)
+      .actorId($cap.getId())
+      .roleCategory(RoleCategory.ADMIN)
+      .roleType(RoleType.ORGANISATION)
+      .roleName("specific-access-approver-admin")
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-admin-mapping.drl:98]: this role requires a worktype of
routine_work

File: src/main/resources/validationrules/publiclaw/publiclaw-caseworker-mapping.drl [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+ */
+rule "publiclaw_specific_access_approver_legal_ops_org_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.PUBLICLAW_WA_1_0.getValue())
+  $cap: CaseWorkerAccessProfile(roleId == "1", serviceCode == "ABA3", !suspended)
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+   attribute.put("workTypes", JacksonUtils.convertObjectIntoJsonNode("decision_making_work,access_requests"));
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-caseworker-mapping.drl:58]: routine_work

File: src/main/resources/validationrules/publiclaw/publiclaw-caseworker-mapping.drl [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+ */
+rule "publiclaw_tribunal_caseworker_org_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.PUBLICLAW_WA_1_0.getValue())
+  $cap: CaseWorkerAccessProfile(roleId == "2", serviceCode == "ABA3", !suspended)
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+   attribute.put("workTypes", JacksonUtils.convertObjectIntoJsonNode("routine_work,hearing_work,applications"));
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-caseworker-mapping.drl:87]: decision_making_work

File: src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+rule "publiclaw_ctsc_team_leader_org_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.PUBLICLAW_WA_1_0.getValue())
+  $cap: CaseWorkerAccessProfile(roleId == "9", serviceCode == "ABA3", !suspended)
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+   attribute.put("workTypes", JacksonUtils.convertObjectIntoJsonNode("routine_work,access_requests,hearing_work,applications"));
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl:91]: routine_work, access_requests

File: src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+
+  insert(
+      RoleAssignment.builder()
+      .actorIdType(ActorIdType.IDAM)
+      .actorId($cap.getId())
+      .roleCategory(RoleCategory.CTSC)
+      .roleType(RoleType.ORGANISATION)
+      .roleName("task-supervisor")
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl:129]: this role requires workType of
routine_work

File: src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+
+  insert(
+      RoleAssignment.builder()
+      .actorIdType(ActorIdType.IDAM)
+      .actorId($cap.getId())
+      .roleCategory(RoleCategory.CTSC)
+      .roleType(RoleType.ORGANISATION)
+      .roleName("case-allocator")
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl:159]: this role requires workType of
routine_work

File: src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+
+  insert(
+      RoleAssignment.builder()
+      .actorIdType(ActorIdType.IDAM)
+      .actorId($cap.getId())
+      .roleCategory(RoleCategory.CTSC)
+      .roleType(RoleType.ORGANISATION)
+      .roleName("specific-access-approver-ctsc")
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl:189]: this role requires workType of
routine_work

File: src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+
+rule "publiclaw_specific_access_approver_ctsc_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.PUBLICLAW_WA_1_0.getValue())
+  $cap: CaseWorkerAccessProfile(roleId =="9", serviceCode == "ABA3", !suspended)
+
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl:181]: regionId is missing

File: src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+
+rule "publiclaw_ctsc_case_allocator_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.PUBLICLAW_WA_1_0.getValue())
+  $cap: CaseWorkerAccessProfile(roleId =="9", serviceCode == "ABA3", !suspended,
+                           caseAllocatorFlag == "Y")
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl:151]: regionId is missing

File: src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+
+rule "publiclaw_ctsc_task_supervisor_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.PUBLICLAW_WA_1_0.getValue())
+  $cap: CaseWorkerAccessProfile(roleId =="9", serviceCode == "ABA3", !suspended,
+                         taskSupervisorFlag == "Y")
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl:121]: regionId is missing

File: src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl (hunk 9 above)
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl:91]: regionId is missing

File: src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+
+rule "publiclaw_ctsc_org_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.PUBLICLAW_WA_1_0.getValue())
+  $cap: CaseWorkerAccessProfile(roleId == "10", serviceCode == "ABA3", !suspended)
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("PUBLICLAW"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($cap.getPrimaryLocationId()));
+   attribute.put("workTypes", JacksonUtils.convertObjectIntoJsonNode("routine_work"));
```
Comment (ConorH994) [inline src/main/resources/validationrules/publiclaw/publiclaw-ctsc-mapping.drl:31]: regionId is missing

**Pull request merged**